#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Modify lib/malloc/malloc.c to make it compile with gcc-3.4

--- bash.orig/lib/malloc/malloc.c 2002-06-21 21:16:49.000000000 +0200 
+++ bash/lib/malloc/malloc.c 2004-07-06 16:17:01.952139000 +0200 
@@ -901,6 +901,7 @@
   if (_malloc_nwatch > 0)
     _malloc_ckwatch (mem, file, line, W_FREE, ubytes);
 #endif
+  return;
 }
 
 static PTR_T
